### PR TITLE
Replace Rufus-scheduler with Heroku scheduler

### PR DESCRIPTION
**Pull request by JayKay24**
#### What does this PR do?

This PR adds replacements for the Rufus cron scheduler with Heroku scheduler.

#### Description
Currently, Rufus-scheduler remains inactive as soon as the Heroku
servers go idle, thus being rendered unable to send the links to
registered users.

This addition replaces the current implementation if Rufus-scheduler with
Heroku scheduler.

### CODE REVIEW

File: app/controllers/link_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class LinkController < ApplicationController
   bing_search_api = BingService::BingWebSearchApi.new
 
-  get '/category_links/:category_id' do
+  get '/category/:category_id/links' do
```
Comment (morriskimani) [inline app/controllers/link_controller.rb:13]: I like the new restful routes. 
Just a small issue: I'd have expected `category` to be plural. Check out the link that I shared on RESTful routes.

File: spec/models/category_model_spec.rb [hunk 2]
```diff
@@ -3,29 +3,18 @@
...
-    DatabaseCleaner.clean
-  end
-
   context 'with valid attributes provided' do
     it 'should create a new category with the logged in user' do
-      expect(user.categories.new(
-               category_name: 'Superman',
-               description: 'A world of heroes'
-             )).to be_valid
+      should validate_presence_of :category_name
```
Comment (morriskimani) [inline spec/models/category_model_spec.rb:8]: No need to change this one now... But in future, prefer `expect` rather than `should`. I think `should` has actually been deprecated. See [this on reasons why](https://github.com/rspec/rspec-expectations/blob/master/Should.md)

File: jobs/link_sender.rb [hunk 3]
```diff
@@ -3,57 +3,61 @@
...
-        canonical_url = result_set[0]['displayUrl'].strip
-        unless result_set[0]['displayUrl'].start_with?('http', 'https')
-          canonical_url = 'https://' + result_set[0]['displayUrl'].strip
-        end
-        link_name = category.links.find_by(
-          link: canonical_url)
-        until link_name.nil?
-          result_set.shuffle!
+module LinkSender
+  def send_links
```
Comment (morriskimani) [inline jobs/link_sender.rb]: IMO, this method is too long. Which means that it's doing too much work. Remember **Single Responsibility Principle**...
Several things are happening here:
- Configuring Pony
- Finding a link that hasn't been sent yet
- Saving and sending the link.
etc.

File: Rakefile [hunk 4]
```diff
@@ -3,11 +3,20 @@ require 'sinatra/activerecord/rake'
...
 
 RSpec::Core::RakeTask.new(:spec) do |t|
   t.verbose = false
 end
 
 task default: :spec
 
-Coveralls::RakeTask.new
\ No newline at end of file
+namespace :mailer do
```
Comment (morriskimani) [inline Rakefile:14]: 👍

**Review by morriskimani — COMMENTED** [review]
A marked improvement on tests. Testing for model validations and more edge cases.

The only major issue on the PR is the `send_links` method which I have flagged. Other than that, looks good!

**Review by morriskimani — APPROVED** [review]
